bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java

```diff
@@ -93,6 +95,10 @@ public void close() {
             session.close();
             session = null;
         }
+
+        if (sessionFuture != null && !sessionFuture.isDone()) {
```

Comment:
@J-N-K I 've an additional question because I've the slight feeling that we still have a potential leak in this class.

The `KodiWebSocketListener` sets the `session = null` in the `onClose()` method.

https://github.com/openhab/openhab-addons/blob/19b5dd357e7b69297101d343c383d3d20d671727/bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java#L158

The `onClose()` method will be called by `onError()` too.  Isn't that a potential problem too? Shouldn't we either cancel the future there too or should we omit setting `session` to `null` anyway. What's your thoughts on that?

Reply:
Randomly seeing this comment, did you reach a conclusion?